reform_cage.py

```diff
@@ -0,0 +1,176 @@
+"""
+From a two-component cage .mol file, returns:
+
+	1) The SMILES code of the building blocks.
+	1.1) The active coordination number of the linkers.
+        1.2) The active functional group.
+        1.3) The number of times a building block has been used in a cage.
+
+	2) The topology of the cage.
+
+        3) Optional Arguments.
+        3.1) Optional (-c): Reform the unrelaxed cage from .mol file.
+        3.2) Optional (-b): Reform the unrelaxed building-blocks (bb).
+
+Usage: $ python reform_cage.py <file.mol>
+
+Author: James T. Pegg
+
+"""
+import argparse
+import stk
+from rdkit.Chem import AllChem
+import operator
+import os
+
+def reform_cage(name, result, deconstructed_topology):
+    """
+    Reform the unrelaxed two-component stk cage.
+
+    """
+    building_blocks = []
+    for k in result:
+      bb_mol = AllChem.MolFromSmiles(k)
+      bb_Hs = AllChem.AddHs(bb_mol)
+      AllChem.EmbedMolecule(bb_Hs)
+      if result[k][0] >= 3:
+          bb_su = stk.StructUnit3(bb_Hs, [result[k][1]])
+          if bb_su not in building_blocks:
+              building_blocks.append(bb_su)
+      else:
+          bb_fg = [result[k][1]]
+          bb_su = stk.StructUnit2(bb_Hs, [result[k][1]])
+          if bb_su not in building_blocks:
+              building_blocks.append(bb_su)
+
+    cage = stk.Cage(building_blocks, deconstructed_topology)
+    cage.write(name+'-out.mol')
+    cage.dump(name+'-out.json')
+    return cage
+
+
+def write_building_blocks(name, result):
+    """
+    Writes the optimised building blocks to .mol file.
+
+    """
```

Comment:
The docstring should have a Parameters Section and a Returns section.
See
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html